dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java

```diff
@@ -53,7 +59,32 @@
     private final ProgressBar progressBar = ProgressBar.getInstance();
     private static Map<String, Object> envMap = null;
 
-    public void synthesizeAndDownloadFeatures(List<String> mavenCoords, String dLocation, MavenRepository repository) throws InstallException {
+    final ExecutorService executor = Executors.newCachedThreadPool();
+
+    public Future<?> esaDownload(String coords, String dLocation, MavenRepository repository) {
+        return executor.submit(() -> {
+            try {
+                synthesizeAndDownload(coords, "esa", dLocation, repository, false);
+            } catch (InstallException e) {
+                e.printStackTrace();
+            }
+            return coords + ".esa";
+        });
+    }
+
+    public Future<?> pomDownload(String coords, String dLocation, MavenRepository repository) {
+        return executor.submit(() -> {
```

Comment:
We can define this task as a Callable<String>() in order to get the hard typed Future<String> suggested above